server/src/main/java/org/elasticsearch/search/retriever/RankDocsRetrieverBuilder.java

```diff
@@ -141,6 +141,9 @@ public void extractToSearchSourceBuilder(SearchSourceBuilder searchSourceBuilder
         if (searchSourceBuilder.size() + searchSourceBuilder.from() > rankDocResults.length) {
             searchSourceBuilder.size(Math.max(0, rankDocResults.length - searchSourceBuilder.from()));
         }
+        if (compoundUsed) {
```

Comment:
Another approach would be to wrap this in a new `AbstractQueryBuilder` that would just override the `extractInnerHitBuilders` to be a no-op and would delegate everything else to the current instance.